robottelo/host_helpers/cli_factory.py

```diff
@@ -1356,3 +1360,93 @@ def setup_cdn_and_custom_repos_content(
             data['lce'] = lce
 
         return data
+
+    def get_available_capsule_port(self, port_pool=None):
+        """returns a list of unused ports dedicated for fake capsules
+        This calls an ss command on the server prompting for a port range. ss
+        returns a list of ports which have a PID assigned (a list of ports
+        which are already used). This function then substracts unavailable ports
+        from the other ones and returns one of available ones randomly.
+
+        :param port_pool: A list of ports used for fake capsules (for RHEL7+: don't
+            forget to set a correct selinux context before otherwise you'll get
+            Connection Refused error)
+
+        :return: Random available port from interval <9091, 9190>.
+        :rtype: int
+        """
+        if port_pool is None:
+            port_pool_range = settings.fake_capsules.port_range
+            if type(port_pool_range) is str:
+                port_pool_range = tuple(port_pool_range.split('-'))
+            if type(port_pool_range) is tuple and len(port_pool_range) == 2:
```

Comment:
use `isinstance` for these checks instead. not only is it a best practice, but it is also faster.